src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -134,26 +169,90 @@ public String processEngineType(final Command requestingCommand, final String ch
+        userProgressStorage.save(userProgress);
+    }
 
-        final String message;
+    private void processM1EngineVolume(final UserProgress userProgress, final Command requestingCommand, final String chatID) throws UtilsborException {
+        if (!UserProgressValidator.validateCommand(Command.M1_GASOLINE_ENGINE_VOLUME, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
+        switch (requestingCommand) {
+            case Command.M1_VOLUME_LESS_1000_CM -> userProgress.setM1Volume(M1EngineVolume.LESS_1000);
+            case Command.M1_VOLUME_BETWEEN_1000_2000_CM ->
+                    userProgress.setM1Volume(M1EngineVolume.BETWEEN_1000_AND_2000);
+            case Command.M1_VOLUME_BETWEEN_2000_3000_CM ->
+                    userProgress.setM1Volume(M1EngineVolume.BETWEEN_2000_AND_3000);
+            case Command.M1_VOLUME_BETWEEN_3000_3500_CM ->
+                    userProgress.setM1Volume(M1EngineVolume.BETWEEN_3000_AND_3500);
+            case M1_VOLUME_MORE_3500_CM -> userProgress.setM1Volume(M1EngineVolume.MORE_3500);
+        }
+        userProgressStorage.save(userProgress);
+    }
+
+    public String processTruckUnitClass(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+
+        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
 
         switch (requestingCommand) {
-            case Command.VOLUME_LESS_1000_CM -> userProgress.setVolumeOfEngine(VolumeOfEngine.LESS_1000);
-            case Command.VOLUME_BETWEEN_1000_2000_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_1000_AND_2000);
-            case Command.VOLUME_BETWEEN_2000_3000_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_2000_AND_3000);
-            case Command.VOLUME_BETWEEN_3000_3500_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_3000_AND_3500);
-            case VOLUME_MORE_3500_CM -> userProgress.setVolumeOfEngine(VolumeOfEngine.MORE_3500);
+            case TRUCK_UNITS_6_CLASS -> userProgress.setTruckUnitType(TruckUnitClass.TRUCK_UNITS_6_CLASS);
+            case TRUCK_UNITS_EXCEPT_6_CLASS -> userProgress.setTruckUnitType(TruckUnitClass.TRUCK_UNITS_EXCEPT_6_CLASS);
         }
         userProgressStorage.save(userProgress);
-        message = messagesCreator.getSummaryAnswer(userProgress);
-        return message;
+        return messagesCreator.getSummaryAnswer(userProgress);
+    }
+
+    public String processTruckUnitWeight(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+        if (!UserProgressValidator.validateCommand(Command.TRUCK_UNIT_WEIGHT, userProgress)) {
```

Comment:
Is command really should be hardocoded?
```suggestion
        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
```

Reply:
In this case yes, thank you for noticing it. 
Sometimes I have a node with different answer. Example: 
![image](https://github.com/user-attachments/assets/eab7f82c-b104-4647-a606-699b8ebe86ee)

But i don't care about user answer - I know that despite user answer he has only one possible next step. 
Example.
![image](https://github.com/user-attachments/assets/0b5fc889-04ba-4055-95c0-ce4089d588f4)
 
Whatever user choose, next step - age auto.
For this cases I decided to unite commands in one common command to avoid useless extra nodes in tree. 
![image](https://github.com/user-attachments/assets/3a928599-c41d-4a0f-af06-54a3d1539203)
This is possibilities for user commands. 

![image](https://github.com/user-attachments/assets/1a9962e4-e71e-4077-b55c-5ed4161c9a91)
This is how it looks in the tree. 

And I mustn't check the particular command, I have to check only does this command belongs to correct group of commands or not and I need to hardcode this command.